### ISSUE TITLE
4.19.1: broken `update-pot` target

### ISSUE THREAD

**Issue by kloczek** [rpm-software-management/rpm#2817]
It would be good to add checking that target to release procedure
```console
+ /usr/bin/cmake --build x86_64-redhat-linux-gnu -j48 --verbose --target update-pot
Change Dir: '/home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu'

Run Build Command(s): /usr/bin/cmake -E env VERBOSE=1 /usr/bin/gmake -f Makefile -j48 update-pot
/usr/bin/cmake -S/home/tkloczko/rpmbuild/BUILD/rpm-4.19.1 -B/home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/gmake  -f CMakeFiles/Makefile2 update-pot
gmake[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu'
/usr/bin/cmake -S/home/tkloczko/rpmbuild/BUILD/rpm-4.19.1 -B/home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu/CMakeFiles 1
/usr/bin/gmake  -f CMakeFiles/Makefile2 po/CMakeFiles/update-pot.dir/all
gmake[2]: Entering directory '/home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu'
/usr/bin/gmake  -f po/CMakeFiles/update-pot.dir/build.make po/CMakeFiles/update-pot.dir/depend
gmake[3]: Entering directory '/home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu'
cd /home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/tkloczko/rpmbuild/BUILD/rpm-4.19.1 /home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/po /home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu /home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu/po /home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu/po/CMakeFiles/update-pot.dir/DependInfo.cmake "--color="
gmake[3]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu'
/usr/bin/gmake  -f po/CMakeFiles/update-pot.dir/build.make po/CMakeFiles/update-pot.dir/build
gmake[3]: Entering directory '/home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu'
[100%] Updating translation source
cd /home/tkloczko/rpmbuild/BUILD/rpm-4.19.1 && /usr/bin/xgettext --package-name=rpm --package-version=4.19.1 --output=/home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/po/rpm.pot --language=C --no-location --keyword=_ --keyword=N_ --files-from=/home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/po/POTFILES.in
/usr/bin/xgettext: error while opening "cliutils.c" for reading: No such file or directory
gmake[3]: *** [po/CMakeFiles/update-pot.dir/build.make:74: po/CMakeFiles/update-pot] Error 1
gmake[3]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu'
gmake[2]: *** [CMakeFiles/Makefile2:1114: po/CMakeFiles/update-pot.dir/all] Error 2
gmake[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu'
gmake[1]: *** [CMakeFiles/Makefile2:1121: po/CMakeFiles/update-pot.dir/rule] Error 2
gmake[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/rpm-4.19.1/x86_64-redhat-linux-gnu'
```

**Comment by kloczek** [comment]
BTW: is it possible to change dist tar ball compression from .bz2 to .xz 🤔 
(xz produces shorter tar balls on archives with text files)
```console
[tkloczko@pers-jacek SOURCES]$ ls -l rpm-4.19.1.tar.xz rpm-4.19.1.tar.bz2
-rw-rw-r-- 1 tkloczko tkloczko 5940447 Dec 14 19:37 rpm-4.19.1.tar.bz2
-rw-rw-r-- 1 tkloczko tkloczko 5449032 Dec 14 19:37 rpm-4.19.1.tar.xz
```

**Comment by pmatilai** [comment]
Why on earth are you specifically running 'update-pot' in the first place? 

And, if you expect somebody to look at this, supply the entire reproducer. Ie, all the cmake commands on a pristine source tarball.

**Comment by kloczek** [comment]
> Why on earth are you specifically running 'update-pot' on a released rpm tarball in the first place?

Because .pot file used during build is ALWAYS not-up-to-date. `update-pot` removes many translated entries which are no longer used (-> installed .mo files are shorter).
As you see in this case what is in .po files bases on not updated files list from which phrases to translate should be taken.
Because .po files updates are not pushed translators are updating outdated list of phrases to translate.

> And, if you expect somebody to look at this, supply the entire reproducer. Ie, all the cmake commands on a pristine source tarball.

In first line of the provided output is ONE command which needs to be executed. It does not need anything build env specific biits.
Please at least try to reproduce that by execute that command in your build env.

**Comment by bracketttc** [comment]
The fix for this is going to be in https://github.com/rpm-software-management/rpm-l10n.
I've opened a PR that allows the target to build on my system: https://github.com/rpm-software-management/rpm-l10n/pull/1

**Comment by kloczek** [comment]
IMO all those changes should be part of rpm repo.
In this repo should be time to time (at least week before release) updated po/rpm.pot file and that .pot files should be registered on one of the i18n translation infras. Than time to time available updates should be pooled.
Really .. please have a look how it is organized by other projects because nothing needs to be kept in separated git repo.

**Comment by bracketttc** [comment]
I'm just making the changes where they are now. I can certainly see moving the CMake code out and file list out.

**Comment by kloczek** [comment]
> I'm just making the changes where they are now. I can certainly see moving the CMake code out and file list out.

It is not possible to tag repo ans subrepo.
At the moment in https://github.com/rpm-software-management/rpm-l10n there is no any tags.

**Comment by pmatilai** [comment]
> In first line of the provided output is ONE command which needs to be executed. It does not need anything build env specific biits.

It's one command run in some unspecific context. You've reported several tickets where you've been using cmake in a very different manner than what we do and expect (it doesn't mean its wrong, just different), and having to interrogate the exact leading steps is waste of everybodys time.

**Comment by pmatilai** [comment]
Actually both of you: what exactly are you doing that you run into this?

Clearly, the paths in the rpm-l10n repo were wrong but the release tarball from ftp.rpm.org has been built locally and for Fedora multiple times by different people and doesn't exhibit any such errors.
Which makes me wonder if you're building from a tag in git or a tarball created by GH based on one. 

The only supported way of building releases is using a tarball from http://ftp.rpm.org/releases/ (as mentioned in https://github.com/rpm-software-management/rpm/releases), and the build commands in INSTALL. Which at this time only includes building with 'make'.

**Comment by kloczek** [comment]
> (..) and for Fedora multiple times by different people and doesn't exhibit any such errors.

Because Fedora build procedure has no `update-pot` target execution.

**Comment by pmatilai** [comment]
Well yes, it's not something you're expected to run on a release tarball. So this is really going in circles, again.

**Comment by kloczek** [comment]
> Well yes, it's not something you're expected to run on a release tarball. So this is really going in circles, again.

MANY projects have not updated .po files (look on https://github.com/mesonbuild/meson/issues/11338)

**Comment by pmatilai** [comment]
> MANY projects have not updated .po files 

I fail to see any relevance to rpm.

In any case, this is supposedly fixed in b4dc72f4b92489f77de9b0ae0bed754875d37ece.

**Comment by eli-schwartz** [comment]
> MANY projects have not updated .po files (look on https://github.com/mesonbuild/meson/issues/11338)

Yes, and we don't believe you should be running update-po either.
